src/net/p2p.rs

```diff
@@ -2789,22 +2828,34 @@ impl PeerNetwork {
                                         block_stats,
                                     )?
                                 {
+                                    let index_block_hash = StacksBlockHeader::make_index_block_hash(
+                                        &parent_consensus_hash,
+                                        &parent_block_hash
+                                    );
+
+                                    if parent_consensus_hash == highest_snapshot.consensus_hash {
+                                        // This parent block was just sortition'ed
```

Comment:
What happens if the antientropy protocol pushes blocks that have just been chosen?

Reply:
It shouldn't, and I've confirmed this when running the node on mainnet.  This condition should make it so the anti-entropy protocol only pushes blocks and microblocks after at least one subsequent sortition has happened (since the block or confirmed microblock stream to be pushed must have a different consensus hash than the burnchain's canonical tip).

Reply:
Right -- my question is, why not? What's the downside of pushing the just-chosen block?

Reply:
It's a DDoS vector.  If we did this, then all nodes you're connected to will all push the block to you at about the same time.  We avoid this today by making it so that all your connected neighbors will only push you a (much smaller) `BlocksAvailable` message when they have obtained a block.  This, in turn, causes you to go and ask the neighbors for the block data if you do not yet have it already.

Reply:
Hilariously, I've witnessed this DDoS-y behavior already.  Remember the network outage from a few weeks ago when we lost a couple blocks for a few hours?  Turns out, a new nodes will have obtained, but not processed, these blocks because the invalid microblock stream that they confirm has been lost forever.  So new nodes don't know that old nodes already have this block, and have simply orphaned it because it confirmed a too-big stream.  As such, a new node will periodically send the invalid block to old nodes that did process the (now-lost) microblock stream over and over because the old nodes report it missing while the new nodes report it as present but unprocessed.

This behavior doesn't pose a systemic risk simply because new nodes booting from this branch will need a new chainstate anyway, so they'll all obtain this unprocessable block and will all believe that their peers already have it.

Reply:
Ah, okay, that makes sense